server/src/main/java/org/elasticsearch/ingest/IngestService.java

```diff
@@ -1238,6 +1239,28 @@ private void executePipelines(
                         return; // document failed!
                     }
 
+                    for (StreamType streamType : StreamType.getEnabledStreamTypesForProject(project)) {
+                        if (streamType.matchesStreamPrefix(newIndex)
+                            && ingestDocument.getIndexHistory().contains(streamType.getStreamName()) == false) {
```

Comment:
I think this would let me through if I rerouted from say `logs` to `logs.abc.def`. That is, it allows me to write to any descendent stream, not just a direct child stream. I assume that's OK, right? The real goal is to prevent things from outside of the stream writing to child streams.